src/charts/onPreprocess/deriveVariables.js

```diff
@@ -0,0 +1,6 @@
+export default function deriveVariables() {
+    this.measure_data.forEach(d => {
+        d.abnormal =
```

Comment:
I think I'd check this the data itself (that is `d[this.config.lln_col]`) when `this.config.normal_range_method === 'LLN-ULN'`, and check against `this._ln` for all the others. That way people can have different data driven upper and lower limits (which is pretty common, I think).

Reply:
Note that I think it's still better to derive `this._ln` and draw the marks based on the median of the values in the data (like it is above)

Reply:
This piece is checking each result against the corresponding normal range in the data when `this.config.normal_range_method === 'LLN-ULN'` as defined in `src/charts/onPreprocess/deriveStatistics`.